--- u-boot-toradex/include/configs/colibri_imx7.h	2019-06-06 09:18:59.933247000 -0300
+++ u-boot-toradex/include/configs/colibri_imx7.h	2019-06-12 15:53:52.444490471 -0300
@@ -163,16 +163,16 @@
 		"fatload ${interface} 0:1 ${loadaddr} " \
 		"${board}/flash_blk.img && source ${loadaddr}\0" \
 	"setup=setenv setupargs " \
-		"console=tty1 console=${console}" \
+		"console=ttyS0 console=${console}" \
 		",${baudrate}n8 ${memargs} consoleblank=0\0" \
 	"setupdate=run setsdupdate || run setusbupdate || run setethupdate\0" \
 	"setusbupdate=usb start && setenv interface usb && " \
 		"fatload ${interface} 0:1 ${loadaddr} " \
 		"${board}/flash_blk.img && source ${loadaddr}\0" \
 	"splashpos=m,m\0" \
-	"videomode=video=ctfb:x:640,y:480,depth:18,pclk:39722,le:48,ri:16,up:33,lo:10,hs:96,vs:2,sync:0,vmode:0\0" \
+	"videomode=video=ctfb:x:800,y:480,depth:18,pclk:30030,le:13,ri:8,up:33,lo:10,hs:40,vs:20,sync:1,vmode:1\0" \
 	"updlevel=2\0" \
-	"vidargs=video=mxsfb:640x480M-16@60"
+	"vidargs=video=mxsfb0:dev=lcd,Chip,if=RGB666 fbmem=32M"
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_LONGHELP
